nx/lib/nx/defn/grad.ex

```diff
@@ -424,6 +424,24 @@ defmodule Nx.Defn.Grad do
     to_grad(x, g, cache)
   end
 
+  defp grad(:concatenate, [tensors, axis], %{shape: ans_shape}, g, cache) do
+    zero_axes = List.duplicate(0, tuple_size(ans_shape))
+    ans_shape_list = Tuple.to_list(ans_shape)
+
+    tensors
+    |> Enum.reduce({0, []}, fn t, {limit, result} ->
+      t_len = elem(t.shape, axis)
+      current_limit = t_len + limit
+      start = List.replace_at(zero_axes, axis, limit)
+      len = List.replace_at(ans_shape_list, axis, t_len)
+
+      result = [{t, Nx.slice(g, start, len)} | result]
```

Comment:
This basically means we travel along the concatenation axis, retrieving the gradient corresponding to the current tensor and associating it to the tensor itself

Reply:
Yup, sounds correct to me!